app/src/androidTest/java/com/example/bigowlapp/activity/SupervisedGroupListActivityTest.java

```diff
@@ -0,0 +1,129 @@
+        testUserData.postValue(testUser);
+
+        testUserSupervisedGroupList = new ArrayList<>();
+        for (int i = 1; i < (int) ((Math.random() * 4) + 4); i++) {
+            User groupSupervisor = new User(
+                    "group00".concat(String.valueOf(i)).concat("supervisor"),
+                    "group00".concat(String.valueOf(i)).concat("fName"),
+                    "group00".concat(String.valueOf(i)).concat("lName"),
+                    "+12300".concat(String.valueOf(i)),
+                    "group00".concat(String.valueOf(i)).concat("@mail.com"),
+                    null
+            );
+            MutableLiveData<User> groupSupervisorData = new MutableLiveData<>(groupSupervisor);
+
+            List<String> groupSupervisedUserId = new ArrayList<>();
+            for (int j = 1; j < (int) ((Math.random() * 2) + 1); j++) {
+                groupSupervisedUserId.add(
+                        "group00".concat(String.valueOf(i)).concat("user00").concat(String.valueOf(j)));
+            }
+            groupSupervisedUserId.add("abc123");
+
+            Group newGroup = new Group(
+                    "group00".concat(String.valueOf(i)),
+                    "groupName00".concat(String.valueOf(i)),
+                    groupSupervisor.getUId(),
+                    groupSupervisedUserId
+            );
+            testUserSupervisedGroupList.add(newGroup);
+
+            when(userRepository.getDocumentByUId(groupSupervisor.getUId(), User.class))
+                    .thenReturn(groupSupervisorData);
+            when(supervisedGroupListViewModel.getSupervisor(groupSupervisor.getUId())).thenReturn(groupSupervisorData);
+        }
+        MutableLiveData<List<Group>> testUserSupervisedGroupListData = new MutableLiveData<>(testUserSupervisedGroupList);
+        testUserSupervisedGroupListData.postValue(testUserSupervisedGroupList);
+
+        when(testFirebaseUser.getUid()).thenReturn("abc123");
+        when(authRepository.getCurrentUser()).thenReturn(testFirebaseUser);
+        when(supervisedGroupListViewModel.getCurrentUserData()).thenReturn(testUserData);
+
+        when(groupRepository.getListOfDocumentByArrayContains(
+                anyString(), anyString(), eq(Group.class)))
+                .thenReturn(testUserSupervisedGroupListData);
+        when(supervisedGroupListViewModel.getSupervisedGroupListData()).thenReturn(testUserSupervisedGroupListData);
+
+        activityRule.getScenario().moveToState(Lifecycle.State.CREATED);
+        activityRule.getScenario().onActivity(activity -> {
+            activity.setSupervisedGroupListViewModel(supervisedGroupListViewModel);
+        });
+        activityRule.getScenario().moveToState(Lifecycle.State.RESUMED);
+    }
+
+    @Test
+    public void activityListOutUserSupervisedGroupsTest() {
+        verify(supervisedGroupListViewModel, atMostOnce()).getCurrentUserData();
+        verify(supervisedGroupListViewModel, atMostOnce()).getSupervisedGroupListData();
```

Comment:
You should use `times(1)` instead of `atMostOnce()`, as atMostOnce will still pass even if the function. Good check though.